node/src/components/chain_synchronizer/operations.rs

```diff
@@ -2240,21 +2247,37 @@ async fn fetch_and_store_deploys<REv>(
     ctx: &ChainSyncContext<'_, REv>,
 ) -> Result<Vec<Deploy>, Error>
 where
-    REv: From<StorageRequest> + From<FetcherRequest<Deploy>> + From<NetworkInfoRequest>,
+    REv: From<StorageRequest>
+        + From<FetcherRequest<Deploy>>
+        + From<NetworkInfoRequest>
+        + From<ControlAnnouncement>,
 {
     let start_instant = Timestamp::now();
 
-    let hashes: Vec<_> = hashes.cloned().collect();
-    let mut deploys: Vec<Deploy> = Vec::with_capacity(hashes.len());
-    let mut stream = futures::stream::iter(hashes)
-        .map(|hash| fetch_and_store_deploy(hash, ctx))
+    let mut ordered_deploys = BTreeMap::new();
```

Comment:
We could use the `mut deploys` from before and insert at an index as well, right?

Reply:
Not sure it makes _too_ much of a difference.  Using `Vec::insert` would require actually constructing a Vec with the right number of entries, which would then all get overwritten (and `insert` can panic, making it one more thing to consider during safety reviews of the code).

If we see performance problems, we could indeed switch to a `Vec` and maybe leverage some unsafe methods, but I'd benchmark before doing something like that.

Reply:
Is there anything wrong with `Vec::sort`?